aea/protocols/base.py

```diff
@@ -114,12 +115,30 @@ def body(self, body: Dict) -> None:
         """
         self._body = body
 
+    @property
+    def dialogue_reference(self) -> Tuple[str, str]:
+        """Get the dialogue_reference of the message."""
+        assert self.is_set("dialogue_reference"), "dialogue_reference is not set."
+        return cast(Tuple[str, str], self.get("dialogue_reference"))
+
     @property
     def message_id(self) -> int:
-        """Get the message id."""
-        assert self.is_set("message_id"), "message_id is not set!"
+        """Get the message_id of the message."""
+        assert self.is_set("message_id"), "message_id is not set."
         return cast(int, self.get("message_id"))
 
+    @property
+    def performative(self) -> Enum:
+        """Get the performative of the message."""
+        assert self.is_set("performative"), "performative is not set."
+        return cast(Enum, self.get("performative"))
+
+    @property
+    def target(self) -> int:
+        """Get the target of the message."""
+        assert self.is_set("target"), "target is not set."
+        return cast(int, self.get("target"))
+
```

Comment:
The base dialogue requires these properties. There is no harm of setting them in the base class as we expect them to be present in all messages anyway as per current generator.

Reply:
OK, so why in the future we don't make the class `Message` less generic? That is, the constructor requires the `performative`, `target` etc.?

Sometimes for a framework, it is good to "take decisions" and force the user to think in the framework's terms.

Reply:
Yes that can be done; although it's probably best to postpone this until later when we have more certainty on the required fields.